### PR TITLE
Rx swift refactor

### PR DESCRIPTION
**Pull request by matiasschwalb**
## Summary ##

Refactored great deal of code using ReactiveSwift + ReactiveCocoa framework

### iPhone X/XS/XR

Simulator Screen Shot - iPhone XR - 2019-06-13 at 10 46 40

### Request Sent

rentRequestWithReactiveSwift

### Book already rented

alreadyRented

### CODE REVIEW

File: WBooks/BookDetailViewController.swift [hunk 1]
```diff
@@ -38,36 +43,67 @@ final class BookDetailViewController: UIViewController {
...
+                }
+                
+                if result.error != nil {
+                    self.rentRequestFailed()
+                }
+            }
             
-            switch rentResult {
+            self.bookDetailViewModel.rent()
+            //self.bookDetailViewModel.finishedRentingSignal.
```
Comment (gmazzei) [inline WBooks/BookDetailViewController.swift]: Remember to delete comments like this.

File: WBooks/BookDetailViewController.swift [hunk 2]
```diff
@@ -5,30 +5,35 @@
...
-        commentList = CommentDB.getCommentsUsingBookID(bookID: bookID)
+    init(withBookDetailViewModel: BookDetailViewModel) {
+        self.bookDetailViewModel = withBookDetailViewModel
         super.init(nibName: "BookDetailViewController", bundle: Bundle.main)
     }
     
     required init?(coder aDecoder: NSCoder) {
-        self.bookID = -1
-        self.commentList = []
+        self.bookDetailViewModel = BookDetailViewModel(book: Book(status: "-1", id: -1, author: "-1", title: "-1", image: "-1", year: "-1", genre: "-1"))
```
Comment (gmazzei) [inline WBooks/BookDetailViewController.swift]: Remove unnecessary `self`'s. 
Also, remove that `Book` with `"-1"` values. You can either create another constructor/initializer that doesn't need a book, or perhaps modify the current one, making `book` an optional parameter.

File: WBooks/BookDetailViewController.swift [hunk 3]
```diff
@@ -5,30 +5,35 @@
...
+        return loadedCommmentsSignalPipe.output
+    }
     
-    private var commentList: [Comment]
+    deinit {
+        loadedCommmentsSignalPipe.input.sendCompleted()
+    }
     
-    let dispatchGroup = DispatchGroup()
+    private var commentList: [Comment] = []
```
Comment (gmazzei) [inline WBooks/BookDetailViewController.swift]: Do you need this variable?

File: WBooks/BookDetailViewModel.swift [hunk 4]
```diff
@@ -8,106 +8,81 @@
...
+    public var rentResult: SignalProducer<Void, RepositoryError> = SignalProducer<Void, RepositoryError> { (_, _) in
+        return
+    }
+
+    private let changeLabelSignalPipe = Signal<String, NoError>.pipe()
+    var changeLabelSignal: Signal<String, NoError> {
+        return changeLabelSignalPipe.output
+    }
+/*
+    private let rentSignalPipe = Signal<Bool, NoError>.pipe()
```
Comment (gmazzei) [inline WBooks/BookDetailViewModel.swift]: Delete this as well.

File: WBooks/LibraryViewModel.swift [hunk 5]
```diff
@@ -14,7 +14,7 @@ import Result
 class LibraryViewModel {
     
     private let mutableBooks = MutableProperty<[Book]>([])
-    public let books: Property<[Book]>
+    public var books: Property<[Book]>
```
Comment (gmazzei) [inline WBooks/LibraryViewModel.swift]: Does this need to be a `var`? I think we could change it to a constant (`let`).

**Comment by matiasschwalb** [comment]
Changes done! Ready for merging

**Pull request merged**